### PR TITLE
PXC-3850 Document that wsrep session variables are updatable through …

### PR DESCRIPTION
**Pull request by patrickbirch**
…SET_VAR() hint (8.0)

modified: source/wsrep-system-index.rst

### CODE REVIEW

**Review by kamil-holubicki — APPROVED** [review]
lgtm

File: doc/source/wsrep-system-index.rst [hunk 1]
```diff
@@ -414,6 +414,25 @@ To avoid deadlock errors,
 set the :variable:`wsrep_sync_wait` variable to ``0``
 if you enable ``wsrep_dirty_reads``.
 
+As of Percona Server 8.0.27, you can update the variable with a `set_var hint <https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var>`__.
+
+.. code-block:: mysql
+
+   mysql> SELECT @@wsrep_dirty_reads;
+   +-----------------------+
+   | @@swrep_dirty_reads   |
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]: Should be `wsrep_dirty_reads`
```suggestion
   | @@wsrep_dirty_reads   |
```

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: doc/source/wsrep-system-index.rst (hunk 1 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Fixed.

File: doc/source/wsrep-system-index.rst [hunk 2]
```diff
@@ -1238,5 +1275,72 @@ is determined by bitmask:
...
+   :dyn: Yes
+   :default: "bytes"
+
+Defines the type of measure for the ``wsrep_trx_fragment_size``. The possible values are: bytes, rows, statements. 
+
+As of *Percona Server for MySQL* 8.0.27, you can update the variable with a `set_var hint <https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var>`__.
+
+.. code-block:: mysql
+
+   mysql> SELECT @@wsrep_trx_fragment_units; SELECT @@wsrep_trx_fragment_size;
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]:
```suggestion
   mysql> SELECT @@wsrep_trx_fragment_unit; SELECT @@wsrep_trx_fragment_size; 
```

File: doc/source/wsrep-system-index.rst [hunk 3]
```diff
@@ -1238,5 +1275,72 @@ is determined by bitmask:
...
+   :dyn: Yes
+   :default: 0
+
+Defines the the streaming replication fragment size. This variable is measured in the units defined by ``wsrep_trx_fragment_unit``. The minimum value is 0 and the maximum value is 2147483647.
+
+As of *Percona Server for MySQL* 8.0.27, you can update the variable with a `set_var hint <https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var>`__.
+
+.. code-block:: mysql
+
+   mysql> SELECT @@@wsrep_trx_fragment_units; SELECT @@wsrep_trx_fragment_size;
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]:
```suggestion
   mysql> SELECT @@@wsrep_trx_fragment_unit; SELECT @@wsrep_trx_fragment_size; 
```

File: doc/source/wsrep-system-index.rst [hunk 4]
```diff
@@ -1238,5 +1275,72 @@ is determined by bitmask:
...
+   +------------------------------+
+   | @@wsrep_trx_fragment_unit    |
+   +==============================+
+   | statements                   |
+   +------------------------------+
+   | @@wsrep_trx_fragment_size    |
+   +------------------------------+
+   | 3                            |
+   +------------------------------+
+   mysql> SELECT /*+ SET_VAR(wsrep_trx_fragment_size=5) */ @@wsrep_trx_fragment_size;
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst:1302]: I think it is also better to tell users that one can use SET_VAR() extension for DMLs as well and can be useful for streaming large statement within a transaction.

```
node1> begin;
Query OK, 0 rows affected (0.00 sec)

node1> INSERT /*+SET_VAR(wsrep_trx_fragment_size = 100)*/ INTO t1 SELECT * FROM t1;
Query OK, 65536 rows affected (15.15 sec)
Records: 65536  Duplicates: 0  Warnings: 0

node1> UPDATE /*+SET_VAR(wsrep_trx_fragment_size = 100)*/ t1 SET i=2;
Query OK, 131072 rows affected (1 min 35.93 sec)
Rows matched: 131072  Changed: 131072  Warnings: 0

node1> DELETE /*+SET_VAR(wsrep_trx_fragment_size = 10000)*/ FROM t1;
Query OK, 131072 rows affected (15.09 sec)



Meanwhile on node2, while UPDATE is running on node1

node2> set session transaction_isolation = 'READ-UNCOMMITTED';
Query OK, 0 rows affected (0.00 sec)

node2> SELECT * FROM t1 LIMIT 10;
+------+
| i    |
+------+
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
|    2 |
+------+
10 rows in set (0.00 sec)
```

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: doc/source/wsrep-system-index.rst (hunk 2 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Fixed.

File: doc/source/wsrep-system-index.rst (hunk 3 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Fixed.

File: doc/source/wsrep-system-index.rst (hunk 4 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst:1302]: Updated examples

File: doc/source/wsrep-system-index.rst [hunk 5]
```diff
@@ -414,6 +414,25 @@ To avoid deadlock errors,
 set the :variable:`wsrep_sync_wait` variable to ``0``
 if you enable ``wsrep_dirty_reads``.
 
+As of Percona Server 8.0.26-16, you can update the variable with a `set_var hint <https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var>`__.
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]: Here and everywhere
```suggestion
As of Percona XtraDB Cluster 8.0.26-16, you can update the variable with a `set_var hint <https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html#optimizer-hints-set-var>`__.
```

File: doc/source/wsrep-system-index.rst [hunk 6]
```diff
@@ -1238,5 +1275,107 @@ is determined by bitmask:
...
+   | @@wsrep_trx_fragment_size    |
+   +==============================+
+   | 5                            |
+   +------------------------------+
+
+You can also use set_var() in a data manipulation language (DML) statement. This ability is useful when streaming large statements within a transaction. 
+
+.. code-block:: mysql
+
+   node1> begin;
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]: Can we use uppercase for all SQL statements?
```suggestion
   node1> BEGIN;
```

File: doc/source/wsrep-system-index.rst [hunk 7]
```diff
@@ -1238,5 +1275,107 @@ is determined by bitmask:
...
+   Query OK, 0 rows affected (0.00 sec)
+
+   node1> INSERT /*+SET_VAR(wsrep_trx_fragment_size = 100)*/ INTO t1 SELECT * FROM t1; 
+   Query OK, 65536 rows affected (15.15 sec)
+   Records: 65536 Duplicates: 0 Warnings: 0
+
+   node1> Update /*+SET_VAR(wsrep_trx_fragment_size = 100)*/ t1 SET i=2;
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]: Same as above
```suggestion
   node1> UPDATE /*+SET_VAR(wsrep_trx_fragment_size = 100)*/ t1 SET i=2;
```

File: doc/source/wsrep-system-index.rst [hunk 8]
```diff
@@ -1238,5 +1275,107 @@ is determined by bitmask:
...
+   Query OK, 131072 rows affected (1 min 35.93 sec)
+   Rows matched: 131072 Changed: 131072 Warnings: 0
+
+   node2> set session transaction_isolation = 'READ-UNCOMMITTED';
```
Comment (venkatesh-prasad-v) [inline doc/source/wsrep-system-index.rst]:
```suggestion
   node2> SET SESSION transaction_isolation = 'READ-UNCOMMITTED';
```

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]

File: doc/source/wsrep-system-index.rst (hunk 5 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Updated.

File: doc/source/wsrep-system-index.rst (hunk 6 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Updated.

File: doc/source/wsrep-system-index.rst (hunk 8 above)
Comment (patrickbirch) [reply doc/source/wsrep-system-index.rst]: Fixed.

**Review by venkatesh-prasad-v — APPROVED** [review]
Thank you for addressing the review comments. This looks good now. Approved!

**Pull request merged**